apps/omg_eth/test/support/dev_geth.ex

```diff
@@ -36,54 +38,88 @@ defmodule OMG.Eth.DevGeth do
     geth = ~s(geth --miner.gastarget 7500000 \
             --nodiscover \
             --maxpeers 0 \
-            --miner.gasprice \"10\" \
-            --datadir /data/ \
+            --miner.gasprice "10" \
             --syncmode 'full' \
             --networkid 1337 \
             --gasprice '1' \
             --keystore #{keystore} \
             --password /tmp/geth-blank-password \
-            --unlock \"0,1\" \
+            --unlock "0,1" \
             --rpc --rpcapi personal,web3,eth,net --rpcaddr 0.0.0.0 --rpcvhosts='*' --rpcport=8545 \
             --ws --wsaddr 0.0.0.0 --wsorigins='*' \
             --allow-insecure-unlock \
             --mine --datadir #{datadir} 2>&1)
-    geth_pid = launch(geth)
+    _pid = launch(geth)
 
     {:ok, :ready} = WaitFor.eth_rpc(20_000)
 
-    on_exit = fn -> stop(geth_pid) end
+    on_exit = fn ->
+      Exexec.run("pkill -9 geth")
+    end
 
     {:ok, on_exit}
   end
 
-  # PRIVATE
+  @impl true
+  def init(cmd) do
+    _ = Logger.debug("Starting geth")
 
-  defp stop(pid) do
-    # NOTE: monitor is required to stop_and_wait, don't know why? `monitor: true` on run doesn't work
-    _ = Process.monitor(pid)
+    {:ok, geth_proc, os_proc} = Exexec.run(cmd, stdout: true)
 
-    {:exit_status, 35_072} = Exexec.stop_and_wait(pid)
-    :ok
+    {:ok, %{geth_proc: geth_proc, os_proc: os_proc, ready?: false}}
   end
 
+  @impl true
+  def handle_info({:stdout, pid, stdout}, %{os_proc: pid} = state) do
+    new_state =
+      if String.contains?(stdout, "IPC endpoint opened") do
+        Map.put(state, :ready?, true)
+      else
+        state
+      end
+
+    if Application.get_env(:omg_eth, :node_logging_in_debug) do
+      _ = Logger.debug("eth node: " <> stdout)
+    end
+
+    {:noreply, new_state}
+  end
+
+  @impl true
+  def handle_call(:ready?, _from, state) do
+    {:reply, state.ready?, state}
+  end
+
+  # PRIVATE
+
   defp launch(cmd) do
-    _ = Logger.debug("Starting geth")
+    {:ok, pid} = start_link(cmd)
 
-    {:ok, geth_proc, _ref, [{:stream, geth_out, _stream_server}]} =
-      Exexec.run(cmd, stdout: :stream, kill_command: "pkill -9 geth")
+    waiting_task_function = fn ->
```

Comment:
I think `function` part in var name in unnecessary. We can take a look on the right side of the assignment and we know it's a function.